src/main/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepository.java

```diff
@@ -263,21 +265,22 @@ public String writeContent(String site, String path, InputStream content) throws
     }
 
     @Override
-    public String createFolder(String site, String path, String name) {
-        logger.debug("Create folder in site '{}' path '{}'", site, path);
+    public String createFolder(String siteId, String path, String name) throws UserNotFoundException, ServiceLayerException {
+        logger.debug("Create folder in site '{}' path '{}'", siteId, path);
         try {
-            StudioBlobStore store = getBlobStore(site, path);
+            StudioBlobStore store = getBlobStore(siteId, path);
             if (store != null) {
-                store.createFolder(site, normalize(path), name);
+                store.createFolder(siteId, normalize(path), name);
             }
-            return localRepositoryV1.createFolder(site, path, name);
+            return localRepositoryV2.createFolder(siteId, path, name);
         } catch (BlobStoreConfigurationMissingException e) {
             logger.debug("No blob store configuration found for site '{}', " +
-                    "will create folder '{}' in the local repository", site, path);
-            return localRepositoryV1.createFolder(site, path, name);
+                    "will create folder '{}' in the local repository", siteId, path);
+            return localRepositoryV2.createFolder(siteId, path, name);
         } catch (Exception e) {
-            logger.error("Failed to create folder in site '{}' path '{}'", site, path, e);
-            return null;
+            logger.error("Failed to create folder in site '{}' path '{}'", siteId, path, e);
+            Path newFolderPath = Paths.get(path, name);
+            throw new ServiceLayerException(format("Error creating new folder at path '{}' site '{}'", newFolderPath, siteId));
```

Comment:
Include the caught Exception `e` in the new ServiceLayerException constructor

Reply:
Updated